examples/by_runtime_api_module/array_management.cu

```diff
@@ -0,0 +1,98 @@
+#include <cuda/api_wrappers.hpp>
+
+#include <iostream>
+
+__global__ void from_texture_to_ptr_kernel3D(cudaTextureObject_t texture_obj, float* ptr, size_t w, size_t h, size_t d) {
+
+	const auto gtidx = threadIdx.x + blockIdx.x * blockDim.x;
+	const auto gtidy = threadIdx.y + blockIdx.y * blockDim.y;
+	const auto gtidz = threadIdx.z + blockIdx.z * blockDim.z;
```

Comment:
Suggest an alternative please

Reply:
Oh, I see what you mean, you're not quite repeating yourself.

Ok, never mind. You could have a `project()` function and and a `global_index_in_linear_grid()` function, but maybe that's overkill.

Reply:
To write this in a shorter way one would need `auto vec_gtid = global_position_in_grid()` to access later the individual coordinates of the thread, e.g. `vec_gtid.x`, `vec_gtid.y` or `vec_gtid[0]`, `vec_gtid[1]`, but this again would require to have an array like class, which is CUDA kernel compatible. Thus I think it would definately be an overkill here, and is maybe better located in your other repository `cuda-kat`.